client/src/components/Overview/AddToCart.jsx

```diff
@@ -1,7 +1,140 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
 
-function AddToCart() {
-  return null;
+// TODO: make Select take all unspecified props from parent
+function Select({
+  label,
+  value,
+  options,
+  defaultSelection,
+  disabled,
+  onChange,
+}) {
+  return (
+    <select aria-label={label} value={value} onChange={onChange}>
+      {options.map((option) => (
+        <option key={option} value={option}>{option}</option>
+      ))}
+    </select>
+  );
 }
 
+Select.propTypes = {
+  label: PropTypes.string.isRequired,
+  value: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.number,
+  ]),
+  options: PropTypes.arrayOf(PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.number,
+  ])),
+  defaultSelection: PropTypes.string,
+  disabled: PropTypes.bool,
+  onChange: PropTypes.func.isRequired,
+};
+
+Select.defaultProps = {
+  options: [],
+  disabled: false,
+};
+
+function AddToCart({ skus }) {
+  const [selectedSize, setSelectedSize] = useState();
+  // TODO: should quantities be state?
+  // It's computed from selectedSize and I only added it to state so
+  // updates to it would trigger a rerender and update
+  // the <Select /> for quantity selection
+  const [quantities, setQuantities] = useState([1]);
+  const [selectedQuantity, setSelectedQuantity] = useState();
+  const [promptForSize, setPromptForSize] = useState(false);
+
+  const availableSkus = Object.values(skus).filter((sku) => sku.quantity > 0);
+  const isInStock = availableSkus.length > 0;
+  let sizes = [];
+
+  if (isInStock) {
+    sizes = availableSkus.map((sku) => sku.size);
+  } else {
+    sizes = ['OUT OF STOCK'];
+  }
+
+  // TODO: should this effect have no subscriptions?
```

Comment:
It only needs to be subscribed if you want it to run EVERY time that skus changes value

Reply:
True. In this case I think it does need to be subscribed since we want to reset the size whenever we change the product or style, which propogates to new skus.